src/app/projects/components/projects/projects.component.ts

```diff
@@ -164,17 +166,16 @@ export class ProjectsComponent implements OnInit, OnDestroy {
     let tmp;
     [this.projectListQuery$, tmp] = this.projectApolloService.getProjects();
     this.subscriptions$.push(tmp.subscribe((projectList) => {
-
       projectList.sort((a, b) => a.name.localeCompare(b.name));
-      projectList.forEach(projectItem => {
-        if (projectItem.createdAt) {
-          projectItem.timeStamp = parseUTC(projectItem.createdAt);
-          const splitDateTime = projectItem.timeStamp.split(',');
-          projectItem.date = splitDateTime[0].trim();
-          projectItem.time = splitDateTime[1];
-        };
-      });
-      this.projectList = projectList.filter(a => a.status != ProjectStatus.IN_DELETION);
+      this.projectList = projectList.map((project) => {
+        const projectItemCopy = jsonCopy(project);
+        projectItemCopy.timeStamp = parseUTC(projectItemCopy.createdAt);
+        const splitDateTime = projectItemCopy.timeStamp.split(',');
+        projectItemCopy.date = splitDateTime[0].trim();
+        projectItemCopy.time = splitDateTime[1];
+        return projectItemCopy;
+      })
+      this.projectList = this.projectList.filter(a => a.status != ProjectStatus.IN_DELETION);
```

Comment:
we first map all and then filter? why not the other way around? 
```suggestion
      this.projectList = projectList.filter(a => a.status != ProjectStatus.IN_DELETION).map((project) => {
        const projectItemCopy = jsonCopy(project);
        projectItemCopy.timeStamp = parseUTC(projectItemCopy.createdAt);
        const splitDateTime = projectItemCopy.timeStamp.split(',');
        projectItemCopy.date = splitDateTime[0].trim();
        projectItemCopy.time = splitDateTime[1];
        return projectItemCopy;
      })
```

Reply:
also i would sort the final list (this.projectList) not the input list

Reply:
yes, filter before map should be the correct behaviour